src/routes/+page.svelte

```diff
@@ -17,13 +29,86 @@
 	let translateX = $derived(percentMoved * ((OVERLAY_OFFSET_LEFT_PERCENT / 100) * innerWidth));
 	let translateY = $derived(percentMoved * (-(OVERLAY_OFFSET_BOTTOM_PERCENT / 100) * innerHeight));
 
+	let sectionRefs: HTMLElement[] = [];
+
+	const intersectionCallback = (entries: IntersectionObserverEntry[]) => {
+		entries.forEach((entry) => {
+			const section = entry.target as HTMLElement;
+
+			// Select all images within the current section
+			const images = section.querySelectorAll<HTMLImageElement>('img');
+
+			if (entry.isIntersecting) {
+				// Show images
+				images.forEach((image) => {
+					image.style.transitionDelay = '0s';
+					image.classList.add('visible');
+				});
+			} else {
+				// Hide images
+				images.forEach((image) => {
+					image.classList.remove('visible');
+				});
+			}
+		});
+	};
+
+	onMount(() => {
+		const observer = new IntersectionObserver(intersectionCallback, {
+			threshold: 0.1 // Trigger when 10% of the section is visible
+		});
+
+		sectionRefs.forEach((section) => observer.observe(section));
+
+		return () => {
+			sectionRefs.forEach((section) => observer.unobserve(section));
+		};
+	});
+
+	let storeSection = (el: HTMLElement) => {
+		if (el && !sectionRefs.includes(el)) {
+			sectionRefs.push(el);
+		}
+	};
+
+	const illustrations: Record<number, SectionIllustrations> = {
+		0: {
+			left: hero0Left,
+			right: hero0Right
+		},
+		1: {
+			right: hero1Right
+		},
+		2: {
+			left: hero2Left
+		},
+		3: {
+			left: hero3Left,
+			right: hero3Right
+		},
+		4: {
+			left: hero4Left,
+			right: hero4Right
+		},
+		5: {
+			left: hero5Left
+		},
+		6: {
+			right: hero6Right
+		},
+		7: {
+			left: hero7Left
+		}
+	};
```

Comment:
just to make this file a touch cleaner, could we export this object from a separate file? (like we do with `dynamics.ts`)